subprojects/GOPATH/src/github.com/microsoft/hivedscheduler/README.md

```diff
@@ -1,24 +1,43 @@
+## Why You Need HiveD
 
-HiveD also optimizes the performance for **[gang scheduling](example/feature/README.md#Gang-Scheduling)**, a typical scheduling requirement for deep learning training job, where all containers should be allocated before the training job can begin. When multiple gang scheduling jobs are competing for the same set of resource, it may lead to resource starvation, where each job only gets partial resource and wait indefinitely. HiveD can schedule all containers within a job in a transactional manner, i.e. all these containers' requirements will be granted or denied as a whole, thus avoids resource starvation and partial resource allocation.
+HiveD provides essential features for deep learning workloads as follows.
 
-HiveD supports multiple job **priorities**. Higher-priority jobs can **[preempt](example/feature/README.md#Intra-VC-Preemption)** lower-priority jobs. HiveD also introduces **[opportunistic job](example/feature/README.md#Opportunistic-Job)**, i.e. the lowest priority jobs that can use other VCs' free resource when possible.
+### Topology-Aware Resource Guarantee
 
-Lastly, HiveD is **fault-tolerant**. It can handle random machine failures and crash recovery.
+The killer feature that distinguishes HiveD is that it provides resource guarantee to each VC, not only in terms of quantity, a numeric value, but also in terms of **[topology](example/feature/README.md#VC-Safety)**, a key requirement of GPU-based training jobs. For example, a traditional scheduler guarantees that a VC can use 8 GPUs. However, it does not know the topology of these 8 GPUs. It is possible that an 8-GPU training job which has to run within a single node, cannot be allocated even if its VC still has 8 free GPUs. This is because these 8 free GPUs may belong to multiple nodes in the physical cluster.
+
+HiveD protects VCs' resources in terms of **cell**, a user-defined resource type that encodes both the quantity and the topology of the GPUs (among other specific hardware configurations, such as GPU type, networking). In the above example, a user can define a cell type of 8-GPU node, and the VC can be assigned one of such cell. Then, HiveD will ensure that *there is always one 8-GPU node available for the VC*, regardless of the other workloads in the cluster.
+
+HiveD allows flexible cell definitions and assignments. For example, users can define cells at multiple topology levels (e.g., PCI-e switch) for finer-grained allocations. Users can also create different cell types for different GPU models (e.g., V100) or networking configuration (e.g., InfiniBand domain) in a heterogeneous cluster.
```

Comment:
>finer-grained [](start = 145, length = 13)

fine-grained is one word, seems no word finer-grained? So "more fine-grained", or just "finer-grain".
Word assignments, allocations, guarantee seems similar, let's only keep one word for avoid confusing.